effects/push-cache/default.nix

```diff
@@ -0,0 +1,160 @@
+{ lib, withSystem, config, ... }:
+let pkgs-x86_64-linux = withSystem "x86_64-linux" ({ pkgs, ... }: pkgs);
+in {
+  imports = [ ../../flake-module.nix ];
+
+  options = {
+    push-cache-effect = {
```

Comment:
We can have a `push-cachix-effect` and `push-attic-effect` instead of the `type` option. That could resolve passing the type in an unobtrusive way.